File: npd2png.py
import glob
import os
import os.path
import sys

from PIL import Image

WIDTH = 320
HEIGHT = 190 #240

def main(argv):
    fileMask = "*.NPD"
    for inputFile in glob.glob(fileMask):
        outputFile = os.path.splitext(inputFile)[0] + ".png"
        print ("Convert %s to %s" % (inputFile, outputFile))
        img = Image.new('RGB', (WIDTH, HEIGHT), "white")
        pixels = img.load()
        with file(inputFile, "rb") as f:
            f.read(14) # Skip header
            row = 0
            while row < HEIGHT:
                data = f.read(WIDTH * 3)
                if not data or len(data) < WIDTH:
                    break
                col = 0
                for i in xrange(0, len(data), 3):
                    pixels[col, row] = tuple(ord(x) for x in data[i:i + 3])
                    col += 1
                row += 1
        img.save(outputFile)
        os.remove(inputFile)

if __name__ == "__main__":
    main(sys.argv[:])

File: npd2png.go
package main

import (
	"image"
	"image/color"
	"image/png"
	"io"
	"log"
	"os"
	"path/filepath"
)

const (
	width  = 320
	height = 190
)

func replaceExt(path string, ext string) string {
	return path[0:len(path)-len(filepath.Ext(path))] + ext
}

func convert(inputFile string, outputFile string) error {
	fileIn, err := os.Open(inputFile)
	if err != nil {
		return err
	}
	defer fileIn.Close()

	fileIn.Seek(14, 0)

	img := image.NewRGBA(image.Rect(0, 0, width, height))

	buffer := make([]byte, width*3)

	for row := 0; row < height; row++ {
		n, err := io.ReadFull(fileIn, buffer)
		if err != nil {
			return err
		}
		if n < width {
			break
		}
		col := 0
		for i := 0; i < len(buffer); i += 3 {
			c := color.RGBA{buffer[i], buffer[i+1], buffer[i+2], 255}
			img.Set(col, row, c)
			col++
		}
	}

	fileOut, err := os.Create(outputFile)
	if err != nil {
		return err
	}
	defer fileOut.Close()

	err = png.Encode(fileOut, img)
	if err != nil {
		return err
	}

	return nil
}

func main() {
	files, err := filepath.Glob("*.NPD")
	if err != nil {
		log.Fatal(err)
	}

	for _, file := range files {
		outFile := replaceExt(file, ".png")
		log.Printf("Convert %s to %s", file, outFile)
		err := convert(file, outFile)
		if err != nil {
			log.Print("Error: ", err.Error())
		} else {
			os.Remove(file)
		}
	}
}
